File: sol/solution/0001-0100/0040. Combination Sum II/Solution.cs
public class Solution
{
    public IList<IList<int>> CombinationSum2(int[] candidates, int target)
    {
        Array.Sort(candidates);
        IList<IList<int>> result = new List<IList<int>>();
        IList<int> combination = new List<int>();
        Backtrack(candidates, target, 0, combination, result);
        return result;
    }
    private void Backtrack(int[] candidates, int target, int start, IList<int> combination, IList<IList<int>> result)
    {
        if (target == 0)
        {
            result.Add(new List<int>(combination));
            return;
        }
        for (int i = start; i < candidates.Length; i++)
        {
            if (i > start && candidates[i] == candidates[i - 1])
            {
                continue;
            }
            if (candidates[i] > target)
            {
                break;
            }
            combination.Add(candidates[i]);
            Backtrack(candidates, target - candidates[i], i + 1, combination, result);
            combination.RemoveAt(combination.Count - 1);
        }
    }
}

File: sol/solution/0001-0100/0040. Combination Sum II/Solution.dart
class Solution {
  void backtrack(List<int> candidates, int target, int start, List<int> combination, List<List<int>> result) {
    if (target == 0) {
      result.add(List<int>.from(combination));
      return;
    }
    for (int i = start; i < candidates.length; i++) {
      if (i > start && candidates[i] == candidates[i - 1]) continue;
      if (candidates[i] > target) break;
      combination.add(candidates[i]);
      backtrack(candidates, target - candidates[i], i + 1, combination, result);
      combination.removeLast();
    }
  }
  List<List<int>> combinationSum2(List<int> candidates, int target) {
    candidates.sort();
    List<List<int>> result = [];
    List<int> combination = [];
    backtrack(candidates, target, 0, combination, result);
    return result;
  }
}